components/common/src/main/java/com/hotels/styx/config/validator/DocumentFormat.java

```diff
@@ -99,33 +103,57 @@ private void validateField(String prefix, Schema.Field field, JsonNode tree) {
 
             validateObject(prefix + name + ".", subObjectSchema, tree.get(name));
 
-        } else if (fieldValue instanceof Schema.ListField) {
-            Schema.ListField listField = (Schema.ListField) fieldValue;
+        } else if (fieldValue instanceof Schema.MapField) {
```

Comment:
This method has become quite long with many levels of nesting. This doesn't need to be fixed now, but we should consider this tech debt to be fixed later.